pkg/agent/attestor/node/node.go

```diff
@@ -306,43 +320,45 @@ func (a *attestor) serverCredFunc(bundle []*x509.Certificate) func() (credential
 
 	// Explicitly not mTLS since we don't have an SVID yet
 	tlsConfig := spiffePeer.NewTLSConfig([]tls.Certificate{})
-	credFunc := func() (credentials.TransportCredentials, error) { return credentials.NewTLS(tlsConfig), nil }
-	return credFunc
+	return func() (credentials.TransportCredentials, error) {
+		return credentials.NewTLS(tlsConfig), nil
+	}
 }
 
-func (a *attestor) parseAttestationResponse(id string, r *node.AttestResponse) ([]*x509.Certificate, *bundleutil.Bundle, error) {
+func (a *attestor) parseAttestationResponse(r *node.AttestResponse) (string, []*x509.Certificate, *bundleutil.Bundle, error) {
 	if r.SvidUpdate == nil {
-		return nil, nil, errors.New("response missing svid update")
+		return "", nil, nil, errors.New("missing svid update")
 	}
-	if len(r.SvidUpdate.Svids) < 1 {
-		return nil, nil, errors.New("no svid received")
+	if len(r.SvidUpdate.Svids) != 1 {
+		return "", nil, nil, fmt.Errorf("expected 1 svid; got %d", len(r.SvidUpdate.Svids))
```

Comment:
Node attestation should only yield a single identity, at least currently. The new behavior below is dependent on that. Since the agent no longer knows its own ID before attestation, it can't use that ID to index into the Svids map, so we rely on there only being a single identity, for the time being.